crates/consensus/src/transaction/envelope.rs

```diff
@@ -65,12 +65,15 @@ pub enum TxEnvelope {
     Legacy(Signed<TxLegacy>),
     /// A [`TxLegacy`] tagged with type 0.
     TaggedLegacy(Signed<TxLegacy>),
-    /// A [`TxEip2930`].
+    /// A [`TxEip2930`] tagged with type 1.
     Eip2930(Signed<TxEip2930>),
-    /// A [`TxEip1559`].
+    /// A [`TxEip1559`] tagged with type 2.
     Eip1559(Signed<TxEip1559>),
-    /// A [`TxEip4844`].
+    /// A [`TxEip4844`] tagged with type 3.
     Eip4844(Signed<TxEip4844>),
+    /// A [`BlobTransaction`], which is a [`crate::TxEip4844`] with a
+    /// [`crate::BlobTransactionSidecar`].
+    Eip4844WithSidecar(Signed<BlobTransaction>),
```

Comment:
hmm, not sure we want two variants for 4844 

this 4844 type is kinda hard to handle, for sending txs we need the sidecar but for historical txs we don't have it.

but I assume we want both formats here
maybe we instead have an enum for 4844?

wdyt @prestwich

Reply:
See https://github.com/alloy-rs/alloy/pull/229/commits/9f433222fa2226cb722b064385cad6fd8ef269e6 for a enum implementation—feels cleaner to me.